libs/test/SmartStoreTest/src/com/salesforce/androidsdk/store/SmartStoreInspectorActivityTest.java

```diff
@@ -281,59 +282,31 @@ private void checkInspectorState(String query, String pageSize,
 
 		// Check results
 		if (expectedResultsAsString == null) {
-			assertNull("Wrong results", getActivity().getLastResults());
+            Assert.assertNull("Wrong results", smartStoreInspectorActivityTestRule.getActivity().getLastResults());
 		} else {
-			assertEquals("Wrong results", expectedResultsAsString,
-					getActivity().getLastResults().toString());
+            Assert.assertEquals("Wrong results", expectedResultsAsString,
+                    smartStoreInspectorActivityTestRule.getActivity().getLastResults().toString());
 		}
 
 		// Check alert
-		assertEquals("Wrong alert title", expectedAlertTitle, getActivity()
+        Assert.assertEquals("Wrong alert title", expectedAlertTitle, smartStoreInspectorActivityTestRule.getActivity()
 				.getLastAlertTitle());
-		String actualAlertMessage = getActivity().getLastAlertMessage();
+		String actualAlertMessage = smartStoreInspectorActivityTestRule.getActivity().getLastAlertMessage();
 		if (expectedAlertMessageSubstring == null
 				|| expectedAlertMessageSubstring.length() == 0) {
-			assertEquals("Wrong alert message", expectedAlertMessageSubstring,
+            Assert.assertEquals("Wrong alert message", expectedAlertMessageSubstring,
 					actualAlertMessage);
 		} else {
-			assertTrue("Wrong alert message",
+            Assert.assertTrue("Wrong alert message",
 					actualAlertMessage.contains(expectedAlertMessageSubstring));
 		}
 	}
-
-    private void clickView(final View v) {
-        try {
-            runTestOnUiThread(new Runnable() {
-                @Override
-                public void run() {
-                    v.performClick();
-                }
-            });
-        } catch (Throwable t) {
-            fail("Failed to click view " + v);
-        }
-    }
     
     private void setText(final int textViewId, final String text) {
         try {
-            runTestOnUiThread(new Runnable() {
-                @Override public void run() {
-                    TextView v = (TextView) getActivity().findViewById(textViewId);
-                    v.setText(text);
-                    if (v instanceof EditText)
-                        ((EditText) v).setSelection(v.getText().length());
-                }
-            });
+            onView(withId(textViewId)).perform(replaceText(text), closeSoftKeyboard());
```

Comment:
Much easier to insert text or perform clicks and other UI actions with views using `Espresso`.